### DISCUSSION TITLE
Using zod to generate table schema

### DISCUSSION THREAD

**Discussion by salehi-hassan** [mswjs/data#265] [Q&A] [answered]
Hey mswjs/data team,

I'm wondering if you have any plan to bring zod validation into table :) Something like this.

```typescript
const PostSchema = {
  dto: z.object({
    id: z.string().uuid().describe('primaryKey'),
    title: z.string(),
    body: z.string().optional().nullable()
  })
}
const PostTable = generateTable(PostSchema.dto)
```

**Comment by kettanaito** [comment]
Hey, @salehi-hassan. 

There is no support for using Zod as the source of truth for the schema. We use our own schema format, although it concerns itself with the data _shape_ without any data type validation on top. 

Frankly, I'm not sure how feasible it'd be to support Zod given that schema format difference. The logic of Data is rooted in the fixed model definition, property proxies that enable relationships, and a few other things specific to Data that Zod won't have. As of now, I would not consider this as a support candidate but you are welcome to pursue adapters between Zod/Data to make things work.

**Reply by salehi-hassan** [reply]
Thanks @kettanaito for the reply. 

True I haven't thought of relationships and the rest of limitation you mentioned. But I like what you did so far, keep up the good work!

**Reply by kettanaito** [reply]
Although I like the idea of reusing the same schema validation logic from zod in Data, one must acknowledge those are tools aimed at different workflows. Data is here to provide swift prototyping and instant API mocking bootstrapping. You absolutely can add an additional validation layer to your Data mocks by yourself (e.g. by running a zod schema against the generated entities).

**Comment by kettanaito** [comment, marked as answer]
Since v1.0, this library _exclusively_ uses Standard Schema-compliant schema libraries, like Zod. Please see the updated `README.md` for more information.
